tutorials/introduction-to-jsosolvers/index.jmd

```diff
@@ -0,0 +1,132 @@
+
+```
+    stats = name_solver(nlp; kwargs...)
+```
+
+where `name_solver` can be `lbfgs`, `R2`, `tron`, or `trunk`, and with
+- `nlp::AbstractNLPModel{T, V}` is an AbstractNLPModel or some specialization, such as an `AbstractNLSModel`;
+- `stats::GenericExecutionStats{T, V}` is a `GenericExecutionStats`, see `SolverCore.jl`.
+
+The keyword arguments may include:
+- `x::V = nlp.meta.x0`: the initial guess.
+- `atol::T = √eps(T)`: absolute tolerance.
+- `rtol::T = √eps(T)`: relative tolerance, the algorithm stops when ‖∇f(xᵏ)‖ ≤ atol + rtol * ‖∇f(x⁰)‖.
+- `max_eval::Int = -1`: maximum number of objective function evaluations.
+- `max_time::Float64 = 30.0`: maximum time limit in seconds.
+- `verbose::Int = 0`: if > 0, display iteration details every `verbose` iteration.
+
+Refer to the documentation of each solver for further details on the available keyword arguments.
+
+## Specialization for nonlinear least-squares
+
+The solvers `tron` and `trunk` both have a specialized implementation for input models of type `AbstractNLSModel`.
+
+The following example illustrate this specialization.
+
+```julia
+using JSOSolvers, ADNLPModels
+f(x) = (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2
+nlp = ADNLPModel(f, [-1.2; 1.0])
+trunk(nlp, verbose = 1)
+```
+
+```julia
+nlp.counters
+```
+
+```julia
+F(x) = [x[1] - 1; 2 * (x[2] - x[1]^2)]
+nls = ADNLSModel(F, [-1.2; 1.0], 2)
+trunk(nls, verbose = 1)
+```
+
+```julia
+nls.counters
+```
+
+## Advanced usage
+
+For advanced usage, first define a `Solver` structure to preallocate the memory used in the algorithm, and then call `solve!`.
+
+```julia
+using JSOSolvers, ADNLPModels
+nlp = ADNLPModel(x -> sum(x.^2), ones(3));
+solver = LBFGSSolver(nlp; mem = 5);
+stats = solve!(solver, nlp)
+```
+
+The following table provides the correspondance between the solvers and the solvers structures:
+
+| Algorithm           | Solver structure |
+| ------------------- | ---------------- |
+| lbfgs               | LBFGSSolver      |
+| R2                  | R2Solver         |
+| tron                | TronSolver       |
+| trunk               | TrunkSolver      |
+| tron (nls-variant)  | TronSolverNLS    |
+| trunk (nls-variant) | TrunkSolverNLS   |
+
+It is also possible to pre-allocate the output structure `stats` and call `solve!(solver, nlp, stats)`.
+```julia
+using JSOSolvers, ADNLPModels, SolverCore
+nlp = ADNLPModel(x -> sum(x.^2), ones(3));
+solver = LBFGSSolver(nlp; mem = 5);
+stats = GenericExecutionStats(nlp)
+solve!(solver, nlp, stats)
+```
+
+## Callback
+
+All the solvers have a callback mechanism called at each iteration.
+The expected signature of the callback is `callback(nlp, solver, stats)`, and its output is ignored.
+Changing any of the input arguments will affect the subsequent iterations.
+In particular, setting `stats.status = :user` will stop the algorithm.
+All relevant information should be available in `nlp` and `solver` see the documentation of each solver for details.
```

Comment:
I am not sure to understand this sentence.

```suggestion
All relevant information should be available in `nlp` and `solver`, see the [documentation of each solver]((https://juliasmoothoptimizers.github.io/JSOSolvers.jl/stable/solvers/)) for details.
```

Reply:
Thanks @paraynaud for the review! I removed this sentence and fixed the rest.